spec/acceptance/class_spec.rb

```diff
@@ -3,22 +3,97 @@
 describe 'agent_upgrade class' do
 
   context 'default parameters' do
-    # Using puppet_apply as a helper
+    before(:all) { setup_puppet }
+    after (:all) { teardown_puppet }
+
     it 'should work idempotently with no errors' do
       pp = <<-EOS
       class { 'agent_upgrade': }
       EOS
 
-      # Run it twice and test for idempotency
+      # TODO: Run it twice and test for idempotency; requires ability to change
+      #       Beaker config to AIO mid-run.
+      apply_manifest(pp, :catch_failures => true)
+      #apply_manifest(pp, :catch_changes  => true)
+    end
+
+    describe package('puppet-agent') do
+      it { is_expected.to be_installed }
+    end
+
+    describe service('puppet') do
+      it { is_expected.to be_enabled }
+      it { is_expected.to be_running }
+    end
+
+    describe service('mcollective') do
+      it { is_expected.to be_enabled }
+      it { is_expected.to be_running }
+    end
+  end
+
+  context 'no services enabled on install' do
+    before(:all) { setup_puppet }
+    after (:all) { teardown_puppet }
+
+    it 'should work idempotently with no errors' do
+      pp = <<-EOS
+      class { 'agent_upgrade': service_names => [] }
+      EOS
+
+      # TODO: Run it twice and test for idempotency; requires ability to change
+      #       Beaker config to AIO mid-run.
       apply_manifest(pp, :catch_failures => true)
-      apply_manifest(pp, :catch_changes  => true)
+      #apply_manifest(pp, :catch_changes  => true)
     end
 
-    describe package('agent_upgrade') do
+    describe package('puppet-agent') do
       it { is_expected.to be_installed }
     end
 
-    describe service('agent_upgrade') do
+    describe service('puppet') do
+      it { is_expected.to_not be_enabled }
+      it { is_expected.to_not be_running }
+    end
+
+    describe service('mcollective') do
+      it { is_expected.to_not be_enabled }
+      it { is_expected.to_not be_running }
+    end
+  end
+
+  context 'with mcollective configured' do
+    before(:all) { setup_puppet }
+    after (:all) { teardown_puppet }
+
+    it 'should work idempotently with no errors' do
+      pp = <<-EOS
```

Comment:
I'm not sure why. It's only needed for this case.

The spec/acceptance model has some funny behavior, like applying a change in one clause and then verifying the effect in others (within a context).